components/cdc/src/endpoint.rs

```diff
@@ -714,30 +714,41 @@ impl<T: 'static + CdcHandle<E>, E: KvEngine, S: StoreRegionMeta> Endpoint<T, E,
         fail_point!("cdc_before_handle_deregister", |_| {});
         match deregister {
             Deregister::Conn(conn_id) => {
-                let conn = self.connections.remove(&conn_id).unwrap();
-                conn.iter_downstreams(|_, region_id, downstream_id, _| {
-                    self.deregister_downstream(region_id, downstream_id, None);
-                });
+                if let Some(conn) = self.connections.remove(&conn_id) {
+                    conn.iter_downstreams(|_, region_id, downstream_id, _| {
+                        self.deregister_downstream(region_id, downstream_id, None);
+                    });
+                } else {
+                    info!("cdc connection already deregistered"; "conn_id" => ?conn_id);
```

Comment:
Shall we print the log here? Looks the connection can only be deregistered once by recv_req exit.